compositor_render/src/state/render_loop.rs

```diff
@@ -91,31 +91,74 @@ pub(super) fn read_outputs(
                     })
                 }
             },
-            None => {
-                let (y, u, v) = RGBColor::BLACK.to_yuv();
-                ctx.wgpu_ctx.utils.fill_r8_with_value(
-                    ctx.wgpu_ctx,
-                    output.output_texture.yuv_textures().plane(0),
-                    y,
-                );
-                ctx.wgpu_ctx.utils.fill_r8_with_value(
-                    ctx.wgpu_ctx,
-                    output.output_texture.yuv_textures().plane(1),
-                    u,
-                );
-                ctx.wgpu_ctx.utils.fill_r8_with_value(
-                    ctx.wgpu_ctx,
-                    output.output_texture.yuv_textures().plane(2),
-                    v,
-                );
-
-                let pending_download = output.output_texture.start_download(ctx.wgpu_ctx);
-                partial_textures.push(PartialOutputFrame::PendingYuvDownload {
-                    output_id: output_id.clone(),
-                    pending_download,
-                    resolution: output.output_texture.resolution().to_owned(),
-                });
-            }
+            None => match output.output_format {
+                OutputFrameFormat::PlanarYuv420Bytes => {
+                    let (y, u, v) = RGBColor::BLACK.to_yuv();
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        output.output_texture.yuv_textures().plane(0),
+                        y,
+                    );
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        output.output_texture.yuv_textures().plane(1),
+                        u,
+                    );
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        output.output_texture.yuv_textures().plane(2),
+                        v,
+                    );
+
+                    let pending_download = output.output_texture.start_download(ctx.wgpu_ctx);
+                    partial_textures.push(PartialOutputFrame::PendingYuvDownload {
+                        output_id: output_id.clone(),
+                        pending_download,
+                        resolution: output.output_texture.resolution().to_owned(),
+                    });
+                }
+                OutputFrameFormat::RgbaWgpuTexture => {
+                    let output_resolution = output.output_texture.resolution();
+                    let wgpu_texture =
```

Comment:
this too low level code to keep here I would:
- Add create_texture method in RGBATexture struct and try to reuse some code
or
- Run `RGBATexture::new()` and extract raw texture from it